src/Microsoft.DotNet.XHarness.Android/AdbRunner.cs

```diff
@@ -129,19 +132,8 @@ public void DumpAdbLog(string outputFilePath, string filterSpec = "")
 
         public void DumpBugReport(string outputFilePath)
         {
-            // give some time for bug report to be available
-            Thread.Sleep(3000);
-
-            var result = RunAdbCommand($"bugreport {outputFilePath}", TimeSpan.FromMinutes(5));
-            if (result.ExitCode != 0)
-            {
-                // Could throw here, but it would tear down a possibly otherwise acceptable execution.
-                _log.LogError($"Error getting ADB bugreport:{Environment.NewLine}{result}");
-            }
-            else
-            {
-                _log.LogInformation($"Wrote ADB bugreport to {outputFilePath}");
-            }
+            var reportManager = AdbReportFactory.CreateReportManager(_log, _api);

```

Comment:
Should we throw when `SetActiveDevice` was not called / `_api` is not set?

Or maybe getting API could be a method and we could do something like `_api ?? GetAPIVersion()` or we can just have `GetAPIVersion` cache the result and we could call it fast many times?

Reply:
In practice we call `SetActiveDevice` at the beginning of each command like `install/test/run`. I may add `GetAPIVersion()` to be sure in case if something changes in this logic.